### ISSUE TITLE
flutter (just audio background) is giving the error for  android 12

### ISSUE THREAD

**Issue by tasaddaq-hussain** [ryanheise/just_audio#699]
**Which API doesn't behave as documented, and how does it misbehave?**
On the all devices (android 9,10,11) my project is working fine but i run on android 12 then its gives an error.
Also when i remove the **(just_audio_background: ^0.0.1-beta.5)** then my app is working fine on android 12.
so this is clear that the issue in this package **(just_audio_background: ^0.0.1-beta.5)**.

**Minimal reproduction project**
I modified the example of just_audio_background.
https://github.com/takuo3/just_audio

**To Reproduce (i.e. user steps, not code)**
1. Debug just_audio / just_audio_background / example in the following repository.
   https://github.com/takuo3/just_audio
2. just pickup the just_audio_background example and try the little code on initState just. 
3. code need to add on initState is 
4. **AudioSource.uri(
  Uri.parse('https://example.com/song1.mp3'),
  tag: MediaItem(
    // Specify a unique ID for each media item:
    id: '1',
    // Metadata to display in the notification:
    album: "Album name",
    title: "Song name",
    artUri: Uri.parse('https://example.com/albumart.jpg'),
  ),
),**
**Note**
remember this the code is working fine on all android else android 12.
5.  

**Error messages**

```
Error: ADB exited with exit code 1
Performing Streamed Install

adb: failed to install C:\flutter_projects\naats_admin\build\app\outputs\flutter-apk\app.apk: Failure [INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl342090594.tmp/base.apk (at Binary XML file line #79): com.ryanheise.audioservice.AudioService: Targeting S+ (version 31 and above) requires that an explicit value for android:exported be defined when intent filters are present]
Error: Failed to install APK again.
Error launching application on sdk gphone64 x86 64.
```

**Expected behavior**
App is not open and not install.

**Desktop (please complete the following information):**
 - OS: [e.g. Window + 10 pro]
 - Browser [e.g. chrome]

**Smartphone (please complete the following information):**
 - Device: [e.g. pixel 3 API 31]
 - OS: [e.g. Android 12]

**Flutter SDK version**
```
C:\Users\tasad>flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.10.4, on Microsoft Windows [Version 10.0.19044.1586], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[√] Chrome - develop for the web
[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 4.1)
[√] VS Code (version 1.66.0)
[√] Connected device (4 available)
[√] HTTP Host Availability

! Doctor found issues in 1 category.
```

**Comment by ryanheise** [comment]
There are additional setup steps required for Android 12 which are not yet documented on that page but if you go to the audio_service page and follow its steps for setting up your manifest, it should work. Thanks for pointing this omission out as I'll need the update just_audio_background's documentation.

**Comment by tasaddaq-hussain** [comment]
@ryanheise 
yes i know also audio_service but am interested with just_audio_background 
so how often will you update  just_audio_background's documentation.
also i shall be thank full to you!

**Comment by ryanheise** [comment]
All I would do is copy the documentation from audio_service since it also applies to just_audio_background. I wasn't suggesting that you use audio_service, only to follow the instructions there for your manifest.

**Comment by tasaddaq-hussain** [comment]
@ryanheise 
I solved this problem by adding this little code into manifest.

**<service android:name="com.ryanheise.audioservice.AudioService"
        android:exported="true" tools:ignore="Instantiatable">
      <intent-filter>
        <action android:name="android.media.browse.MediaBrowserService" />
      </intent-filter>
    </service>

    
    <receiver android:name="com.ryanheise.audioservice.MediaButtonReceiver"
        android:exported="true" tools:ignore="Instantiatable">
      <intent-filter>
        <action android:name="android.intent.action.MEDIA_BUTTON" />
      </intent-filter>
    </receiver> **

**Note**
Add this line to solve the problem:
**android:exported="true" tools:ignore="Instantiatable">**

Thanks

**Comment by ryanheise** [comment]
Yep that's what I was referring you to use.

**Comment by tasaddaq-hussain** [comment]
Thank you so much

**Comment by ryanheise** [comment]
I am reopening until I copy across the documentation.

**Comment by ryanheise** [comment]
I'll close now that I have copied across the relevant parts of the README from audio_service regarding exported=true etc.
